datatypes/src/primitives/time_step.rs

```diff
@@ -0,0 +1,936 @@
+use std::{cmp::max, ops::Add};
+
+use chrono::{Datelike, Duration, NaiveDate};
+use error::Error::NoDateTimeValid;
+use serde::{Deserialize, Serialize};
+
+use crate::error;
+use crate::primitives::TimeInstance;
+use crate::util::Result;
+
+use super::TimeInterval;
+
+/// A time granularity.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub enum TimeGranularity {
+    Seconds,
```

Comment:
Since we have millisecond granularity on `TimeInterval`, we could also list `Milliseconds` here.

Reply:
287487edfa00569018e10f2fa49e3e1933e9708f